### PR TITLE
Various menu improvements

### LINKED ISSUE #981: [Contextual Menu] Create UI and design tokens
```[tasklist]
### Tasks
- [x] Create UI based on existing component
- [x] Design tokens for SL light mode (based on contextual tokens)
- [x] Design tokens for SL dark mode (based on contextual tokens)
- [ ] update theme build for the new tokens (jeroen)
```

### LINKED ISSUE #1137: [Contextual Menu] Finetuning code
```[tasklist]
### Tasks
- [ ] Finetune contextual menu based on UI design
- [ ] Apply new tokens based on new structure (contextual tokens)
```

### LINKED ISSUE #1427: [menu button] remove "single" and "multiple" select variants
in storybook there are "single" and "multiple" select stories. This is not an option in Figma (yet) we don't really know what the use case for these is, how it differs from a regular select / combobox.
We need to have some background info such as use cases/ usages/ patterns to be able to write proper documentation if we actually need those variants.
https://storybook.sanomalearning.design/?path=/story/components-menu-button--single-select
https://storybook.sanomalearning.design/?path=/story/components-menu-button--multi-select
```[tasklist]
### Tasks
- [x] Discuss
- [ ] remove multi and single select property menu button
- [ ] Review documentation, is there anything to update?
```

### PR DESCRIPTION
**Pull request by jpzwarte**
## Menu
- Refactor styling to use the new contextual tokens
- Fix the "safe triangle" not working properly in some cases
- Add animations when the menu is shown
- Improve menu stories and add "All" story
- Various small fixes

## Menu button
- Remove single/multi select in `<sl-menu-button>`

## Shared
- Change popover positioning to use `inset` rather than `translate` (related to the safe triangle fixes)

## Dialog
- Add "Lazy" story to dialog to show how to lazy initialize a dialog

Fixes #981 
Fixes #1137 
Fixes #1427

### CODE REVIEW

File: packages/components/menu/src/menu-button.ts [hunk 1]
```diff
@@ -53,19 +54,9 @@ export class MenuButton extends ScopedElementsMixin(LitElement) {
...
-  @property({ attribute: false }) pluralize?: (count: number) => string;
-
   /** The position of the menu relative to the button. */
   @property() position?: PopoverPosition;
 
-  /** The text representing the selected menuitem(s) that will be shown in the menu button after selection. */
-  @state() selected?: string;
-
-  /** Determines whether if and how many menu items can be selected. */
-  @property() selects?: 'single' | 'multiple';
```
Comment (anna-lach) [inline packages/components/menu/src/menu-button.ts:67]: Can you please also update the documentation (eg. examples) of the `menu` and change the status there from `planned` to `new`? :)

File: packages/components/menu/src/menu-button.stories.ts [hunk 2]
```diff
@@ -168,7 +180,7 @@ export const WithAvatar: Story = {
   }
 };
 
-export const All: Story = {
+export const Combination: Story = {
```
Comment (anna-lach) [inline packages/components/menu/src/menu-button.stories.ts:183]: It looks like the example needs to be updated? :)

image

File: packages/components/menu/src/menu-button.stories.ts [hunk 3]
```diff
@@ -121,41 +132,42 @@ export const Disabled: Story = {
...
-export const MultiSelect: Story = {
+export const Text: Story = {
   args: {
-    body: html`<span slot="button">Lists</span>`,
+    ...Basic.args,
+    body: html`<span slot="button">Settings</span>`
+  }
+};
+
+export const Submenu: Story = {
```
Comment (anna-lach) [inline packages/components/menu/src/menu-button.stories.ts:152]: In the example scrollbar is shown when I hover over the menu
image

File: packages/components/menu/src/menu-button.stories.ts [hunk 4]
```diff
@@ -217,3 +229,118 @@ export const All: Story = {
...
 };
+
+export const All: Story = {
+  render: () => html`
+    <style>
+      #root-inner {
+        align-items: center;
+        display: inline-grid;
+        gap: 1rem;
+        grid-template-columns: auto auto auto auto auto;
```
Comment (anna-lach) [inline packages/components/menu/src/menu-button.stories.ts:240]: Maybe `repeat(5, auto)`?

File: packages/components/menu/src/menu.scss [hunk 5]
```diff
@@ -1,77 +1,126 @@
...
+  gap: var(--sl-space-new-xs);
   margin: 0;
   max-inline-size: var(--sl-menu-max-inline-size, auto);
-  min-inline-size: var(--sl-menu-min-inline-size, fit-content);
+  min-inline-size: var(--sl-menu-min-inline-size, min-content);
   opacity: 0;
   overflow: visible auto;
-  padding-block: var(--_padding-block);
+  padding: var(--sl-space-new-md);
   padding-inline: 0;
```
Comment (anna-lach) [inline packages/components/menu/src/menu.scss:17]: We have a `padding` the line above, so we don't need this padding-inline? Or maybe above it should be `padding-block`?

File: packages/components/menu/src/menu-item.scss [hunk 6]
```diff
@@ -1,98 +1,110 @@
...
-  --_border-radius: var(--sl-border-radius-popover-default);
-  --_cursor: pointer;
-  --_focus-outline: var(--sl-color-focusring-default) solid var(--sl-border-width-focusring-default);
-  --_gap: 0.25rem;
-  --_indent: calc(var(--sl-menu-item-indent, 0) * 26px + 6px);
-  --_gap-shortcut: 0.5rem;
-  --_gap-sub-menu-indicator: 1rem;
-  --_padding: 6px;
-  --_submenu-offset: -5px;
-
```
Comment (anna-lach) [inline packages/components/menu/src/menu-item.scss:12]: We are not setting font-size anymore? It inherits from the body, yes?

File: packages/components/menu/src/menu.ts [hunk 7]
```diff
@@ -188,7 +196,7 @@ export class Menu extends LitElement {
 
   #onToggle(event: Event): void {
     if ((event as ToggleEvent).newState === 'closed' && this.anchorElement instanceof MenuItem) {
-      this.anchorElement.focus();
+      // this.anchorElement.focus();
```
Comment (anna-lach) [inline packages/components/menu/src/menu.ts:199]: Do we want to keep this comment?

File: packages/components/menu/src/menu-item.ts [hunk 8]
```diff
@@ -195,6 +200,7 @@ export class MenuItem extends ScopedElementsMixin(LitElement) {
     this.submenu?.showPopover();
 
     if (focus) {
+      // this.submenu?.querySelector<HTMLElement>('sl-menu-item:not(disabled)')?.focus();
```
Comment (anna-lach) [inline packages/components/menu/src/menu-item.ts:203]: Do we want to keep this comment?

File: packages/components/menu/src/menu.stories.ts [hunk 9]
```diff
@@ -89,8 +128,7 @@ export const Grouped: Story = {
         <sl-icon name="far-gear"></sl-icon>
         Settings
       </sl-menu-item>
-      <hr />
-      <sl-menu-item-group heading="Design System">
```
Comment (anna-lach) [inline packages/components/menu/src/menu.stories.ts:93]: In figma we have examples with group title/heading:

image


Can you please also add an example with group heading?

File: packages/components/menu/src/menu-button.stories.ts (hunk 2 above)
Comment (jpzwarte) [reply packages/components/menu/src/menu-button.stories.ts:183]: Fixed

File: packages/components/menu/src/menu-item.scss (hunk 6 above)
Comment (jpzwarte) [reply packages/components/menu/src/menu-item.scss:12]: That's something i need to talk to @RoaldBoerema about

File: packages/components/menu/src/menu.ts (hunk 7 above)
Comment (jpzwarte) [reply packages/components/menu/src/menu.ts:199]: Nope

File: packages/components/menu/src/menu-item.ts (hunk 8 above)
Comment (jpzwarte) [reply packages/components/menu/src/menu-item.ts:203]: Nope

File: packages/components/menu/src/menu-button.stories.ts (hunk 3 above)
Comment (jpzwarte) [reply packages/components/menu/src/menu-button.stories.ts:152]: Fixed

**Review by Diaan — APPROVED** [review]

File: packages/components/menu/src/menu-button.stories.ts [hunk 10]
```diff
@@ -169,51 +181,116 @@ export const WithAvatar: Story = {
 };
 
 export const All: Story = {
```
Comment (anna-lach) [inline packages/components/menu/src/menu-button.stories.ts:183]: It looks like the focus jumps to another menu button when I'm using a keyboard to open a menu, it only works fine when I'm opening the first menu. I think the focus should stay on the menu button which has opened the menu.
image

Comment (jpzwarte) [reply packages/components/menu/src/menu-button.stories.ts:183]: Fixed

**Review by anna-lach — APPROVED** [review]

**Pull request merged**